### PR TITLE
VRF-950-Fix-SP-5-setLinkToken-Is-Publicly-Exposed

### PR DESCRIPTION
**Pull request by leeyikjiun**
- Remove `setLinkToken` in `VRFV2PlusWrapperConsumerBase` and made link immutable. Link is now retrieved from `VRFV2PlusWrapper` on constructor.

- Made link immutable in `VRFV2PlusWrapper` and check if address is non zero on constructor.

### CODE REVIEW

File: contracts/test/v0.8/foundry/vrf/VRFV2PlusWrapper.t.sol [hunk 1]
```diff
@@ -136,28 +136,14 @@ contract VRFV2PlusWrapperTest is BaseTest {
     new VRFV2PlusWrapper(address(0), address(0), address(0));
   }
 
-  function testSetLinkAndLinkNativeFeed() public {
-    VRFV2PlusWrapper wrapper = new VRFV2PlusWrapper(address(0), address(0), address(s_testCoordinator));
+  function testSetLinkNativeFeed() public {
+    VRFV2PlusWrapper wrapper = new VRFV2PlusWrapper(address(s_linkToken), address(0), address(s_testCoordinator));
```
Comment (ibrajer) [inline contracts/test/v0.8/foundry/vrf/VRFV2PlusWrapper.t.sol:140]: Optional: add a check that the constructor will revert with `ZeroAddress` if the LINK token address is set to zero.

File: contracts/src/v0.8/vrf/dev/interfaces/IVRFV2PlusWrapper.sol [hunk 2]
```diff
@@ -87,4 +87,6 @@ interface IVRFV2PlusWrapper {
     uint32 _numWords,
     bytes memory extraArgs
   ) external payable returns (uint256 requestId);
+
+  function link() external returns (address);
```
Comment (ibrajer) [inline contracts/src/v0.8/vrf/dev/interfaces/IVRFV2PlusWrapper.sol:91]: Why is this not a `view` function?

File: contracts/test/v0.8/foundry/vrf/VRFV2PlusWrapper.t.sol (hunk 1 above)
Comment (leeyikjiun) [reply contracts/test/v0.8/foundry/vrf/VRFV2PlusWrapper.t.sol:140]: Right. I wanted to add that test but I forgot about it. Thanks for catching this.

Comment (leeyikjiun) [reply contracts/test/v0.8/foundry/vrf/VRFV2PlusWrapper.t.sol:140]: Oh, it's sort of already tested through this [PR](https://github.com/smartcontractkit/chainlink/pull/12483)

**Review by ibrajer — APPROVED** [review]
LGTM

**Pull request merged**